### PR TITLE
Respect the cardinality from the Croissant 1.0 specs.

### PR DESCRIPTION
**Pull request by marcenacp**

### CODE REVIEW

File: python/mlcroissant/mlcroissant/_src/core/json_ld.py [hunk 1]
```diff
@@ -114,6 +114,20 @@ def unbox_singleton_list(d: Any):
     return d
 
 
+def box_singleton_list(element: Any) -> list[Any] | None:
+    """Returns the elemtn always boxed inside a list.
```
Comment (ccl-core) [inline python/mlcroissant/mlcroissant/_src/core/json_ld.py:118]: element

inside a list -> inside of a list

Can you also add an example of such property? I assume things such as containedIn woulf fix into this box?

File: python/mlcroissant/mlcroissant/_src/operation_graph/operations/filter.py [hunk 2]
```diff
@@ -13,6 +15,23 @@
 from mlcroissant._src.structure_graph.nodes.file_set import FileSet
 
 
+@functools.cache
+def _memoized_regex(pattern: str) -> re.Pattern:
+    regex = fnmatch.translate(pattern)
+    return re.compile(regex)
+
+
+def match(patterns: list[str] | None, path: pathlib.PurePath) -> bool:
```
Comment (ccl-core) [inline python/mlcroissant/mlcroissant/_src/operation_graph/operations/filter.py:24]: matches_path ?Maybe use a different name for this function? Match is a bit general maybe.

**Review by ccl-core — APPROVED** [review]
Nice!

File: python/mlcroissant/mlcroissant/_src/operation_graph/operations/filter.py (hunk 2 above)
Comment (marcenacp) [reply python/mlcroissant/mlcroissant/_src/operation_graph/operations/filter.py:24]: Done.

File: python/mlcroissant/mlcroissant/_src/core/json_ld.py (hunk 1 above)
Comment (marcenacp) [reply python/mlcroissant/mlcroissant/_src/core/json_ld.py]: Done. The docstring is better now, thanks!

**Pull request merged**